dev/InfoBar/TestUI/Themes/InfoBarStyles.xaml

```diff
@@ -133,48 +145,109 @@
+
+                            <Rectangle x:Name="StandardColor" 
+                                       HorizontalAlignment="Stretch" 
+                                       VerticalAlignment="Stretch" 
+                                       Grid.ColumnSpan="6" 
+                                       Grid.RowSpan="3">
                                 <Rectangle.Fill>
-                                    <SolidColorBrush x:Name="StandardBrush" Color="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=(local:InfoBar.StatusColor)}"/>
+                                    <SolidColorBrush x:Name="StandardBrush" 
+                                                     Color="{Binding RelativeSource={RelativeSource TemplatedParent},Path=(local:InfoBar.StatusColor)}"/>
                                 </Rectangle.Fill>
                             </Rectangle>
-                            <Rectangle x:Name="UserColor" Width="50" Grid.Column="0" HorizontalAlignment="Left" VerticalAlignment="Stretch">
+
+                            <Rectangle x:Name="UserColor" 
+                                       HorizontalAlignment="Stretch" 
+                                       VerticalAlignment="Stretch" 
+                                       Grid.ColumnSpan="6"
+                                       Grid.RowSpan="3">
                                 <Rectangle.Fill>
-                                    <SolidColorBrush x:Name="UserBrush" Color="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=(local:InfoBar.StatusColor)}"/>
+                                    <SolidColorBrush x:Name="UserBrush" 
+                                                     Color="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=(local:InfoBar.StatusColor)}"/>
                                 </Rectangle.Fill>
                             </Rectangle>
 
-                            <IconSourceElement x:Name="StandardIcon" Grid.Column="1" HorizontalAlignment="Right" />
-                            <IconSourceElement x:Name="UserIcon" IconSource="{TemplateBinding IconSource}" Grid.Column="1" HorizontalAlignment="Right" />
+                            <Grid x:Name="ContentRootGrid"
+                                  Padding="12, 0, 12, 0">
+                                <Grid.ColumnDefinitions>
+                                    <ColumnDefinition Width="Auto"/>
+                                    <ColumnDefinition Width="*"/>
+                                    <ColumnDefinition Width="Auto"/>
+                                </Grid.ColumnDefinitions>
+
+                                <IconSourceElement x:Name="StandardIcon"
+                                                   Grid.Column="0"/>
+
+                                <IconSourceElement x:Name="UserIcon" 
+                                                   IconSource="{TemplateBinding IconSource}"
+                                                   Grid.Column="0"/>
```

Comment:
Why are there two IconSourceElements?

Reply:
One IconSourceElement hosts the user's custom IconSource and the other IconSourceElement hosts the severity level's preset IconSource. They will collapse/become visible depending on if the user has set their own custom IconSource. Having the two elements lets me specify the behavior that if the user has both a specified severity level and a specified custom iconsource, then the custom iconsource will override the severity level's iconsource.